packages/web/src/components/premium-content-purchase-modal/PremiumContentPurchaseModal.tsx

```diff
@@ -224,8 +237,13 @@ export const PremiumContentPurchaseModal = () => {
     dispatch(cleanupUSDCRecovery())
   }, [onClosed, dispatch])
 
-  if (track && !isValidTrack) {
+  if (
+    !track ||
+    !purchaseConditions ||
+    !(isValidDownloadGatedTrack || isValidStreamGatedTrack)
```

Comment:
small thing i noticed is that on mobile we also check `!isUSDCEnabled`. probs not needed tho. the whole flag can probs be removed at this point

Reply:
good catch